extensions/big-data-cluster/src/bigDataCluster/dialog/bdcDashboardOverviewPage.ts

```diff
@@ -205,6 +205,18 @@ export class BdcDashboardOverviewPage {
 		}
 
 		this.endpointsRowContainer.clearItems();
+
+		// Sort the endpoints. The sort method is that SQL Server Master is first - followed by all
+		// others in alphabetical order by endpoint
+		const sqlServerMasterEndpoints = endpoints.filter(e => e.name === Endpoint.sqlServerMaster);
+		endpoints = endpoints.filter(e => e.name !== Endpoint.sqlServerMaster)
+			.sort((e1, e2) => {
+				if (e1.endpoint < e2.endpoint) { return -1; }
+				if (e1.endpoint > e2.endpoint) { return 1; }
+				return 0;
+			});
+		endpoints.unshift(...sqlServerMasterEndpoints);
```

Comment:
only do this if it exists (handle potential null ref just in case)

Reply:
If what exists? I'm filtering so if for some reason there aren't any master endpoints this would just add an empty array which is harmless. And endpointsArray is going to be the same thing - it'll never be undefined - just empty.

Reply:
Apologies, I had misread and thought it was a find (and hence could be undefined). This is fine in that case.

Reply:
Cool, yeah I had started off with a find but I figured if for some crazy reason there's multiple SQL Server Master nodes returned we might as well put all of them at the top.